#!/usr/bin/env python
PACKAGE = "myrmex_gripper_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# the three values at the end: default, min, max
gen.add("k", int_t, 0, "Slope", 1600, 50, 5000)
gen.add("Ki", double_t, 0, "PID integral part", 9.0, 0.01, 10.0)
gen.add("Kp", double_t, 0, "PID proportional part", 1.9, 0.01, 3.0)
gen.add("f_target", int_t, 0, "target for sum of forces", 2000, 500, 40000)
gen.add("goal_maintain",   bool_t,   0, "maintain force once reached?",  True)
gen.add("force_threshold", int_t, 0, "Force noise threshold", 1555, 500, 40000)

exit(gen.generate(PACKAGE, "myrmex_gripper_controller", "MyrmexControllerDR"))